core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/TupleExprBuilder.java

```diff
@@ -2613,7 +2611,7 @@ public void meet(Sum node) throws VisitorException {
 
 		private void meetAggregate(AggregateOperator node) {
 			if (node.equals(operator)) {
-				node.getParentNode().replaceChildNode(node, replacement);
+				node.getParentNode().replaceChildNode(node, replacement.clone());
```

Comment:
@jeenbroekstra I think that might not have previously had any tests covering this code here, otherwise the assertion should have failed I think.

Reply:
Peculiar - we have tests for SPARQL queries with aggregate operators I'm sure. Not sure why this code path was not hit by them though.